src/coreclr/jit/importercalls.cpp

```diff
@@ -8606,6 +8618,12 @@ void Compiler::impDevirtualizeCall(GenTreeCall*            call,
                                                                                      &rootCompiler->info.compMethodInfo->args);
 #endif // DEBUG
 
+    if ((JitConfig.JitEnableGenericVirtualDevirtualization() == 0) && call->IsGenericVirtual(this))
+    {
+        JITDUMP("\nimpDevirtualizeCall: generic virtual devirtualization disabled\n");
+        return;
+    }
```

Comment:
This seems like something that is the responsibility of `IsDevirtualizationCandidate` to check.